cfme/tests/generic_objects/test_instances.py

```diff
@@ -284,3 +339,35 @@ def cleanup():
     ).success
 
     assert gen_obj_def_import_export.exists
+
+
+@pytest.mark.meta(automates=[1729341, 1743266])
+def test_generic_object_with_service_button(appliance, generic_object, button_with_dialog):
+    """
+    Bugzilla:
+        1729341
+        1743266
+
+    Polarion:
+        assignee: jdupuy
+        initialEstimate: 1/6h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.10
+        casecomponent: GenericObjects
+    """
+    # add generic object to service
+    myservice = MyService(appliance, name=generic_object.associations.get("services")[0].name)
+    myservice.add_resource_generic_object(generic_object)
+
+    # now navigate to the details of the generic_object
+    view = navigate_to(generic_object, "MyServiceDetails")
+    view.toolbar.button(button_with_dialog.name).custom_button.click()
+    view = generic_object.create_view(TextInputDialogView, wait=10)
+    view.service_name.fill("Custom Button Execute")
+    wait_for(lambda: not view.submit.disabled, timeout="10s")
+    view.submit.click()
+
+    # now for the actual test, make sure after hitting submit we're on the correct page
+    view = generic_object.create_view(MyServiceGenericObjectInstanceView, wait=10)
```

Comment:
Did you want to assert something here?

Reply:
I got a comment from @digitronik `as we are asserting view; It will take care by wait`, so having an assert that the view is displayed after this line would be moot cause it wouldn't be hit if the view wasn't displayed.

Reply:
Does `create_view` do that assertion? I remember creating view for v2v tests which has no way to assert, only navigating to a view would validate that. Maybe that has changed since I last used it.

Reply:
The test will fail with a TimedOutError if it exceeds the wait, we will have to nest it in a try/except if we want it to `Fail`

Reply:
For a test case it would make more sense if it fails as `AssertionError` than `TimedOutError` in my opinion. Cause in a test run, things time out all the time, and we usually see it as `Error` not `Failure` and we really care about reporting `Pass` or `Fail` , hence I'd suggest to have some mechanism to make it pass/fail.

Reply:
Yep, I agree, in this case it is the main assertion of the test so we should use a try/except, updating now.